### PR TITLE
Add obelix linter job

### PR DESCRIPTION
**Pull request by 0xpanoramix**
Signed-off-by: Luca Georges Francois <luca.georges-francois@epitech.eu>

# Description

This PR adds a new job for running the golintci linter.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have tested this code
- [X] I have added sufficient documentation in the code

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  pull_request:
+    branches: ["master", "develop"]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  linter:
+    name: Run linter in Idefix and Panoramix
```
Comment (TomChv) [inline .github/workflows/lint.yml]: on?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+name: Lint Obelix
```
Comment (TomChv) [inline .github/workflows/lint.yml]: Just Lint, I think we should also do it on backend, create an issue please

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+on:
+  push:
+    branches: ["master", "develop"]
+  pull_request:
+    branches: ["master", "develop"]
```
Comment (TomChv) [inline .github/workflows/lint.yml]: Don't forget main

**Review by TomChv — CHANGES_REQUESTED** [review]
Few comments, create a strategy was a good idea ;)

File: .github/workflows/lint.yml (hunk 1 above)
Comment (0xpanoramix) [reply .github/workflows/lint.yml]: Nice catch, fixed in https://github.com/Cleopha/ifttt-like/pull/30/commits/91e3d7ee7d3b21bf95a5f309f9c27414aa071348

File: .github/workflows/lint.yml (hunk 2 above)
Comment (0xpanoramix) [reply .github/workflows/lint.yml]: Got it, done in https://github.com/Cleopha/ifttt-like/pull/30/commits/91e3d7ee7d3b21bf95a5f309f9c27414aa071348

File: .github/workflows/lint.yml (hunk 3 above)
Comment (0xpanoramix) [reply .github/workflows/lint.yml]: Added in https://github.com/Cleopha/ifttt-like/pull/30/commits/91e3d7ee7d3b21bf95a5f309f9c27414aa071348

**Review by TomChv — APPROVED** [review]
LGTM, fix lint error and go

**Comment by TomChv** [comment]
Force merged, fix linter later

**Pull request merged**